libs/blocks/section-metadata/section-metadata.js

```diff
@@ -71,6 +77,37 @@ export const getMetadata = (el) => [...el.childNodes].reduce((rdx, row) => {
   return rdx;
 }, {});
 
+function handleShowMoreButton(section, showMoreBtn, seeMoreText) {
+  if (!showMoreBtn) return;
+  const iconSpan = createTag('span', {
+    class: 'show-more-icon',
+    'aria-hidden': 'true',
+  }, `${ADD_MORE_ICON}`);
+  showMoreBtn.appendChild(iconSpan);
+  showMoreBtn.setAttribute('aria-label', seeMoreText);
+
+  showMoreBtn.addEventListener('click', () => {
+    section.classList.add('show-all');
+    section.querySelector('.show-more-button').remove(); // Remove the entire show-more-button div
+  });
+}
+
+async function handleCollapseSection(text, section) {
+  if (!text || !section) return;
+  const blocks = section.querySelectorAll('div:not(:last-child)');
+  const existingShowMoreButton = section.querySelector('.show-more-button');
+  if (text === 'on' && blocks.length > 3 && !existingShowMoreButton) {
```

Comment:
Can we maybe avoid unnecessary indentation like this: 
```suggestion
  const blocks = section?.querySelectorAll('div:not(:last-child)');
  const existingShowMoreButton = section?.querySelector('.show-more-button');
  if (text !== 'on' || !blocks.length <= 3 || existingShowMoreButton) return;
```